nb_train.py

```diff
@@ -0,0 +1,90 @@
+import pandas as pd
+import string
+import re
+import nltk
+import random
+import pickle
+from sklearn.model_selection import train_test_split
+from nltk.corpus import stopwords
+from zipfile import ZipFile
+
```

Comment:
Overall comment on structure:

You asked when to use a class and when to use functions. Roughly, you should use a class whenever you have some amount of "state", aka data, that you need to store and access within your program. The benefit of encapsulating the data within a class is that you can restrict the way in which the rest of your program will access that data. This lets you ensure that all the ways in which your program might access that data make sense. It also makes it easier to understand what is going on.

Some examples of what "state" might be:

-- the current status of a tic-tac-toe game, if you're writing program that plays tic tac toe
-- the state of a file, if you're writing a program that reads and writes to a file
-- the state of a connection to a server, if you're writing a server

In our case, the "state" we need to store is the model itself. You're already using the `nltk.NaiveBayesClassifier` class to encapsulate the model's internal state, but I think it is worth defining our own class that wraps the `nltk.NaiveBayesClassifier` so we can standardize the operations that can be performed on the model within our program. This will also help us later when you have other classifiers; so long as they all have the same methods, we can write other parts of the code to operate generically over any classifier.

So I propose you create your own NaiveBayesClassifier class, with the following methods:

-- an `__init__` method[1] that takes in a trained `nltk.NaiveBayesClassifier` object. It then stores that object as a class member.
-- an  classmethod[2] `from_pickled` method that takes in a file path and returns an instance of your class initialized with the `nltk.NaiveBayesClassifier` object obtained by unpickling the bytes in the file. I realize this is a bit of an advanced concept so let me know if you get stuck here.
-- a `test` method that takes in a set of test data, evaluates the model on the test data, and returns the accuracy.
-- a `predict` method that takes in a string and returns a prediction for the sentiment of that string.
-- a `store` method that takes in a file path and stores the model at that file path using the `pickle` library.


[1]: https://www.pythontutorial.net/python-oop/python-__init__/
[2]: https://www.tutorialsteacher.com/python/classmethod-decorator